keras/backend/cntk_backend.py

```diff
@@ -1515,27 +1515,17 @@ def conv2d(x, kernel, strides=(1, 1), padding='valid',
     x = _preprocess_conv2d_input(x, data_format)
     kernel = _preprocess_conv2d_kernel(kernel, data_format)
     padding = _preprocess_border_mode(padding)
-    if dilation_rate == (1, 1):
-        strides = (1,) + strides
-        x = C.convolution(
-            kernel,
-            x,
-            strides,
-            auto_padding=[
-                False,
-                padding,
-                padding])
-    else:
-        assert dilation_rate[0] == dilation_rate[1]
-        assert strides == (1, 1), 'Invalid strides for dilated convolution'
```

Comment:
CNTK supports both scenarios, so remove the checks here. As TensorFlow dilated conv doesn't support ```strides != (1, 1)```, so I don't update the ```Conv2D``` API docs, which still say
```Currently, specifying any `dilation_rate` value != 1 is incompatible with specifying any stride value != 1```.